### PR TITLE
Form pages

### PR DESCRIPTION
**Pull request by CornerstoneII**
In this Milestone:
We created and built some forms using nearly-pure HTML and then graduating to using the helper methods that Rails provides.

## Built With

- Ruby 2.7x
- Ruby On Rails 6 or higher
- SQlite3

## Methods Used in user_controllers
- new
- create
- update
- edit
- show

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    respond_to do |format|
+      if @user.save
+        format.html { redirect_to @user, notice: 'User was successfully created.' }
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
```
Comment (peterrobert) [inline app/controllers/users_controller.rb]: If the user is saved, It should redirect_to new_user_path. 
user_create

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  def update
+    respond_to do |format|
+      if @user.update(user_params)
+        format.html { redirect_to @user, notice: 'User was successfully updated.' }
+      else
+        format.html { render :edit }
+      end
+    end
+  end
+
```
Comment (peterrobert) [inline app/controllers/users_controller.rb]: If the user is updated, redirect_to edit_user_path

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    @users = User.all
+  end
+
+  def show; end
+
+  def new
+    @user = User.new
+  end
+
+  def edit; end
```
Comment (peterrobert) [inline app/controllers/users_controller.rb]: In your edit method, you should find the user with the params[id].
user_route2

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED :recycle: :stop_sign: 

HELLO @CornerstoneII  AND @sinansevgi  VERY NICE PROGRESS, YOU HAVE DONE SUCH A GOOD JOB. I LOVE THE WAY YOU HAVE IMPLEMENTED YOUR ROUTING :+1: :100: 

HOWEVER, THERE ARE A FEW CHANGES TO BE MADE KINDLY WORK ON THE FOLLOWING.

- [x] You should have a commented out HTML form. If you check [here](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms) there are two exercises that you need to do. That is create an HTML form and comment it out and also form_tag

kindly work on those changes and submit your work for another review.

### HAPPY CODING :computer:

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  def show; end
+
+  def new
+    @user = User.new
+  end
+
+  def edit
+    @user = User.find(params[:id])
```
Comment (trekab) [inline app/controllers/users_controller.rb:15]: This is a repetition since you already run `set_user` as a before action for edit.

**Review by trekab — APPROVED** [review]
### Approved! :tada: 
Well done, team! Your work with displaying error messages for your forms is commendable. Keep up the good work! Consider removing the minor repetition indicated in the comment in the code below before merging your PR.

_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[Portfolio](https://trekab.github.io/portfolio/) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

**Pull request merged**